convert.lua

```diff
@@ -113,14 +113,22 @@ function is_past_convert_msg(key, pkt_num)
            pkt_num > convert_end_pkt_num[key]
 end
 
--- Assume for the time being that any TCP stream that starts with a SYN with
--- a payload carries Convert data.
+-- Assume that any TCP stream that starts with a SYN with a payload carries
+-- Convert data.
 function is_convert_syn()
     return tostring(tcp_syn_f()) == '1' and
            tostring(tcp_ack_f()) == '0' and
            tostring(tcp_len_f()) ~= '0'
 end
 
+-- Assume that any TCP stream that has an ACK with a payload carries Convert
+-- data.
```

Comment:
If we exclude exceptions (SYN / some RST), all TCP packets have the ACK flag and many of them carry data. In other words, if you don't limit here to the first data packet of a stream, you cannot assume that. If you do that, all packets carrying data will be considered as being part of the convert communication channel.

Reply:
Ack